src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,42 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.AgeException;
+import core.basesyntax.exception.LoginException;
+import core.basesyntax.exception.PasswordException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        String login = user.getLogin();
+        String password = user.getPassword();
+        Integer age = user.getAge();
+        if (login == null) {
+            throw new LoginException("Username must not be empty");
```

Comment:
same for other places

```suggestion
            throw new RegistrationException("Username must not be empty");
```